datafusion/common/src/scalar.rs

```diff
@@ -1360,8 +1396,10 @@ impl ScalarValue {
             // explicitly enumerate unsupported types so newly added
             // types must be aknowledged
             DataType::Float16
-            | DataType::Time32(_)
-            | DataType::Time64(_)
+            | DataType::Time32(TimeUnit::Microsecond)
```

Comment:
super-nitpick: to avoid confusion, we could add a comment here saying that the time units are the reason why this is unsupported.